src/coreclr/jit/decomposelongs.cpp

```diff
@@ -1905,24 +1905,19 @@ GenTree* DecomposeLongs::StoreNodeToVar(LIR::Use& use)
 
     if (user->OperGet() == GT_STORE_LCL_VAR)
     {
-        // If parent is already a STORE_LCL_VAR, we can skip it if
-        // it is already marked as lvIsMultiRegRet.
-        unsigned varNum = user->AsLclVarCommon()->GetLclNum();
-        if (m_compiler->lvaTable[varNum].lvIsMultiRegRet)
-        {
-            return tree->gtNext;
-        }
-        else if (!m_compiler->lvaTable[varNum].lvPromoted)
-        {
-            // If var wasn't promoted, we can just set lvIsMultiRegRet.
-            m_compiler->lvaTable[varNum].lvIsMultiRegRet = true;
-            return tree->gtNext;
-        }
+        // If parent is already a STORE_LCL_VAR, just mark it lvIsMultiRegRet.
+        m_compiler->lvaGetDesc(user->AsLclVar())->lvIsMultiRegRet = true;
```

Comment:
Is there any significance in checking for `m_compiler->lvaEnregMultiRegVars` before setting `lvIsMultiRegRet`?

Reply:
`lvIsMultiRegRet` must always be set for a local defined from a multi-reg source, regardless of whether we are going to enregister the individual fields or not.

(Though, at this point, there are no correctness dependencies on it being set, just one assert in codegen)